vendor/github.com/moby/buildkit/frontend/dockerfile/instructions/bflag.go

```diff
@@ -108,6 +108,15 @@ func (fl *Flag) IsUsed() bool {
 	return false
 }
 
+// Used returns a slice of flag names that are set
+func (bf *BFlags) Used() []string {
+	used := make([]string, 0, len(bf.used))
+	for f := range bf.used {
+		used = append(used, f)
+	}
+	return used
+}
```

Comment:
Temporary patch; perhaps there's a cleaner approach for this, but thought I'd open a PR to discuss the best approach.